server/mdm/apple/apple_bm.go

```diff
@@ -22,15 +24,37 @@ func SetABMTokenMetadata(
 	abmToken *fleet.ABMToken,
 	depStorage storage.AllDEPStorage,
 	ds fleet.Datastore,
-	logger kitlog.Logger) error {
-
+	logger kitlog.Logger,
+) error {
 	decryptedToken, err := assets.ABMToken(ctx, ds, abmToken.OrganizationName)
 	if err != nil {
-		return err
+		return ctxerr.Wrap(ctx, err, "getting ABM token")
 	}
 
+	return SetDecryptedABMTokenMetadata(ctx, abmToken, decryptedToken, depStorage, ds, logger)
+}
+
+func SetDecryptedABMTokenMetadata(
+	ctx context.Context,
+	abmToken *fleet.ABMToken,
+	decryptedToken *depclient.OAuth1Tokens,
+	depStorage storage.AllDEPStorage,
+	ds fleet.Datastore,
+	logger kitlog.Logger,
+) error {
 	depClient := NewDEPClient(depStorage, ds, logger)
-	res, err := depClient.AccountDetail(ctx, abmToken.OrganizationName)
+
+	orgName := abmToken.OrganizationName
+	if orgName == "" {
+		// Then this is a newly uploaded token, which will not be found in the datastore when
+		// RetrieveAuthTokens tries to find it. Set the token in the context so that downstream we
+		// know it's not in the datastore.
+		ctx = context.WithValue(ctx, ABMTokenContextKey, decryptedToken)
```

Comment:
What we typically do to keep track of things that can be stored in the context (and to make a nicer high-level API to use it) is to create a sub-package in `server/contextx/`. I can take care of that refactoring when addressing the cron ticket, though (assuming I'm the one grabbing it!).

Reply:
@mna TIL, nice! I took a swing at it: https://github.com/fleetdm/fleet/pull/21419/files#diff-a0c9522177b546ffab8f5a8f351a55b85ea9d2c4cf96df723a9abb623bb41f88